optaweb-employee-rostering-frontend/src/ui/components/DataTable.tsx

```diff
@@ -194,21 +203,19 @@ export const DataTable = (props: PaginationData<any> & RouteComponentProps & Dat
           <Tr role="row">
             {props.columns.map((header, index) => {
               const sorterProps = (header.sorter) ? {
-                sort: {
-                  sortBy: {
-                    index: props.sortByIndex,
-                    direction: (props.isReversed ? 'desc' : 'asc') as 'desc' | 'asc',
-                  },
-                  onSort: () => props.onSorterChange(index),
-                  columnIndex: index,
-                },
-              } : {};
+                sortBy: {
+                  index: props.sortByIndex,
+                  direction: (props.isReversed ? 'desc' : 'asc'),
```

Comment:
This doesn't cause a type error? When I tried it, it type resolved to 'string' in my IDE, but 'direction' only accept type 'desc' | 'asc'.

Reply:
No, IntelliJ IDEA 2020.3 doesn't complain and `npm run typecheck` passes without any errors.